tests/integration/test_client.py

```diff
@@ -589,6 +590,124 @@ async def test_batch(client: Client, table: TableName) -> None:
     assert len([item async for item in client.query(table, HashKey("h", "h"))]) == 0
 
 
+async def test_transact_write_items(
+    client: Client, table: TableName, is_dynalite_emulator: bool
+) -> None:
+    if is_dynalite_emulator:
+        pytest.xfail("transaction not supported by dynalite yet")
+
+    with pytest.raises(errors.TransactionCanceled):
+        await client.transact_write_items(items=[])
+
+    with pytest.raises(errors.TransactionCanceled):
+        puts = [Put(table=table, item={"h": "h", "r": str(i)}) for i in range(26)]
+        await client.transact_write_items(items=puts)
+
+    puts = [
+        Put(table=table, item={"h": "h", "r": str(i), "s": "initial"}) for i in range(2)
+    ]
+    response = await client.transact_write_items(items=puts)
+    assert not response
+    assert len([item async for item in client.query(table, HashKey("h", "h"))]) == 2
+
+    updates = [
+        Update(
+            table=table,
+            key={"h": "h", "r": str(i)},
+            expression=F("s").set(f"changed"),
+        )
+        for i in range(2)
+    ]
+    response = await client.transact_write_items(items=updates)
+    assert not response
+    items = [item["s"] async for item in client.query(table, HashKey("h", "h"))]
+    assert items == ["changed", "changed"]
+
+    deletes = [
+        Delete(
+            table=table,
+            key={"h": "h", "r": str(i)},
+        )
+        for i in range(2)
+    ]
+    response = await client.transact_write_items(items=deletes)
+    assert not response
+    assert len([item async for item in client.query(table, HashKey("h", "h"))]) == 0
```

Comment:
It's great to see all these tests, but could you please split them up into individual tests all testing a specific part of the API? This test for example checks input validation (which could be a single test function with `pytest.mark.parametrize` for the two cases) as well as the `Put`, `Update` and `Delete` operations which should also be their own test.

The shorter the test the easier it is to see what is tested and also failures in one part of the test don't hide failures in later parts.